pkg/files/s3_integration_test.go

```diff
@@ -0,0 +1,329 @@
+// Vikunja is a to-do list application to facilitate your life.
+// Copyright 2018-present Vikunja and contributors. All rights reserved.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package files
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"code.vikunja.io/api/pkg/config"
+	"code.vikunja.io/api/pkg/db"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestFileStorageIntegration tests end-to-end file storage and retrieval
+// with S3/MinIO storage backend. This test specifically validates S3 functionality
+// and will fail if S3 is not properly configured.
+func TestFileStorageIntegration(t *testing.T) {
+	if os.Getenv("VIKUNJA_TESTS_S3") != "1" {
```

Comment:
please use `config.FilesType.GetString() != "s3"` and then in the ci set the `VIKUNJA_TESTS_USE_CONFIG` environment variable